src/components/Header/WalletMenu.tsx

```diff
@@ -0,0 +1,340 @@
+`
+
+const SlippageItem = styled.div`
+  display: flex;
+  padding: 8px 16px;
+`
+const SlippageContainer = styled.div`
+  cursor: pointer;
+  display: flex;
+  padding: 8px;
+  border-radius: 8px;
+  gap: 8px;
+  align-items: center;
+  background: ${({theme}) => theme.dark.grey3};
+`
+const SubMenuContainer = styled.div`
+  & > a > div {
+    padding-left: 42px;
+  }
+`
+
+const UnstyledAnchorTag = styled.a`
+  color: unset;
+  text-decoration: none;
+`
+
+const MenuLink = ({background, link, onClick, children}: {background?: string; link?: string; onClick?: Function; children: any}) => {
+  if (link) {
+    return (
+      <UnstyledAnchorTag href={link} target="_blank">
+        <MenuItem>{children}</MenuItem>
+      </UnstyledAnchorTag>
+    )
+  }
+  return (
+    <MenuItem onClick={() => (onClick ? onClick() : null)} background={background}>
+      {children}
+    </MenuItem>
+  )
+}
+
+const ModalContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  padding: 24px;
+  flex: 1;
+  position: relative;
+  gap: 4px;
+  background: ${({theme}) => theme.dark.background};
+`
+
+const CloseIcon = styled.div`
+  position: absolute;
+  top: 24px;
+  right: 24px;
+  :hover {
+    cursor: pointer;
+    opacity: 0.7;
+  }
+`
+
+const StyledNumericalInputContainer = styled(NumericalInputContainer)`
+  border: 0;
+  background: #10131d;
+  border-radius: 4px;
+  overflow: hidden;
+  margin-bottom: 0px;
+`
+
+export default function WalletMenu() {
+  const {deactivate, account} = useActiveWeb3React()
+  const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false)
+  const [isSubMenuOpen, setIsSubMenuOpen] = useState<boolean>(false)
+  const [isSlippageModalOpen, setIsSlippageModalOpen] = useState<boolean>(false)
+  const {setSlippageValue} = useBuildState()
+  const {onSetSlippage} = useBuildActionHandlers()
+
+  const menuRef = useRef<HTMLDivElement>(null)
+
+  const disconnectWallet = () => {
+    deactivate()
+    localStorage.setItem('disconnected', 'true')
+    window.location.reload()
+  }
+
+  const showMenu = (val: boolean) => {
+    setIsMenuOpen(val)
+  }
+
+  const handleResetTxnSettings = useCallback(
+    (e: any) => {
+      onSetSlippage(DefaultTxnSettings.DEFAULT_SLIPPAGE)
+    },
+    [onSetSlippage],
+  )
+
+  useEffect(() => {
+    const closeMenu = (event: MouseEvent | TouchEvent) => {
+      // Check if the clicked element is outside the div
+      if (menuRef.current && !menuRef.current.contains(event.target as Node) && event.target !== document.getElementById('showButton')) {
+        setIsMenuOpen(false)
+      }
+    }
+
+    document.addEventListener('click', closeMenu)
+
+    return () => {
+      document.removeEventListener('click', closeMenu)
+    }
+  }, [])
+
+  useEffect(() => {
+    const fetchSlippage = async () => {
+      if (account) {
+        const storedSlippage = localStorage.getItem(`slippage`)
+        if (storedSlippage) {
+          onSetSlippage(storedSlippage || '1')
```

Comment:
use DefaultTxnSettings.DEFAULT_SLIPPAGE instead of '1'

```suggestion
          onSetSlippage(storedSlippage || DefaultTxnSettings.DEFAULT_SLIPPAGE)
```